astroid/rebuilder.py

```diff
@@ -873,9 +876,9 @@ def visit_arguments(self, node: "ast.arguments", parent: NodeNG) -> nodes.Argume
         )
         # save argument names in locals:
         if vararg:
-            newnode.parent.set_local(vararg, newnode)
+            newnode.parent.set_local(vararg, newnode)  # type: ignore[union-attr] # newnode is initialized with a parent
         if kwarg:
-            newnode.parent.set_local(kwarg, newnode)
+            newnode.parent.set_local(kwarg, newnode)  # type: ignore[union-attr] # newnode is initialized with a parent
```

Comment:
An idea: Instead of `type: ignore`, try `assert newnode.parent` above `if vararg`. As it's always initialized with a parent, this should be a no-op.